event_sale_registration_multi_qty/models/sale_order.py

```diff
@@ -11,32 +10,26 @@ class SaleOrderLine(models.Model):
     _inherit = 'sale.order.line'
 
     @api.multi
-    def _update_registrations(self, confirm=True, registration_data=None):
-        """ Writes registrations """
+    def _update_registrations(self, confirm=True, cancel_to_draft=False,
+                              registration_data=None):
+        """Update registrations on events with multi qty enabled"""
         Registration = self.env['event.registration']
-        registrations = Registration.search([
-            ('sale_order_line_id', 'in', self.ids)
-        ])
         for so_line in self.filtered('event_id'):
             if not so_line.event_id.registration_multi_qty:
-                super(SaleOrderLine, so_line)._update_registrations()
+                super(SaleOrderLine, so_line)._update_registrations(
+                    confirm=confirm, cancel_to_draft=cancel_to_draft,
+                    registration_data=registration_data)
                 continue
-            existing_registrations = registrations.filtered(
-                lambda self: self.sale_order_line_id.id == so_line.id)
-            if confirm:
-                existing_registrations.filtered(
-                    lambda self: self.state != 'open'
-                ).confirm_registration()
-            else:
-                existing_registrations.filtered(
-                    lambda self: self.state == 'cancel').do_draft()
-            registration = {}
-            if registration_data:
-                registration = registration_data.pop()
-            if not existing_registrations:
-                registration['sale_order_line_id'] = so_line
-                registration['qty'] = int(so_line.product_uom_qty)
-                Registration.with_context(
-                    registration_force_draft=True).create(
-                    Registration._prepare_attendee_values(registration))
-        return True
+            product_uom_qty = so_line.product_uom_qty
+            # Set temporarily the order line to one avoiding to create more
+            # than one registration
+            so_line.product_uom_qty = 1
+            super(SaleOrderLine, so_line)._update_registrations(
+                confirm=confirm, cancel_to_draft=cancel_to_draft,
+                registration_data=registration_data)
+            # Set the so line qty back and set the registration qty
+            so_line.product_uom_qty = product_uom_qty
+            registration = Registration.search([
+                ('sale_order_line_id', '=', so_line.id)
+            ])
+            registration.qty = int(product_uom_qty)
```

Comment:
Isn't this duplicated with https://github.com/OCA/event/pull/139/commits/cb46d203fe4872ba9eb12c286f739d4e437ebb20#diff-191ffd8beda0d9fbfe9d0541696ca4c8R15?

Reply:
No, because this method updates an existing attendee since the other creates it

Reply:
Can't this be unified then?

Reply:
On a second thought, the line in the _prepare it's not necessary due to the last refactor of this :) Sending patch